boards/arduino/nano_matter/doc/index.rst

```diff
@@ -0,0 +1,116 @@
+.. zephyr:board:: arduino_nano_matter
+
+Overview
+********
+
+The Nano Matter merges the well-known Arduino way of making complex technology more accessible with the powerful MGM240S from Silicon Labs, to bring Matter closer to the maker world, in one of the smallest form factors in the market.
+
+It enables 802.15.4 (Thread®) and Bluetooth® Low Energy connectivity, to interact with Matter-compatible devices with a user-friendly software layer ready for quick prototyping.
+
+The Nano Matter features a compact and efficient architecture powered by the MGM240S (32-bit Arm® Cortex®-M33) from Silicon Labs, a high-performance wireless module optimized for the needs of battery and line-powered IoT devices for 2.4 GHz mesh networks.
+
+Hardware
+********
+
+- MGM240SD22VNA2 Mighty Gecko SiP
+- CPU core: ARM Cortex®-M33 with FPU
+- Flash memory: 1536 kB
+- RAM: 256 kB
+- Transmit power: up to +20 dBm
+- Operation frequency: 2.4 GHz
+- Crystals for LFXO (32.768 kHz) and HFXO (39 MHz).
+- User RGB LED
+- User button
+
+For more information about the EFR32MG24 SoC and the Arduino Nano Matter, refer to these
+documents:
+
+- `MGM240S Website`_
+- `EFR32MG24 Datasheet`_
+- `EFR32xG24 Reference Manual`_
+- `Nano Matter User Manual`_
+
+Supported Features
+==================
+
+.. zephyr:board-supported-hw::
+
+Connections and IOs
+===================
+
+In the following table, the column **Name** contains Pin names. For example, PA2
+means Pin number 2 on PORTA, as used in the board's datasheets and manuals.
+
++-------+-------------+------------------+
+| Name  | Function    | Usage            |
++=======+=============+==================+
+| PC1   | GPIO        | LED0             |
++-------+-------------+------------------+
+| PC2   | GPIO        | LED1             |
++-------+-------------+------------------+
+| PC3   | GPIO        | LED2             |
++-------+-------------+------------------+
+| PA0   | GPIO        | Button           |
++-------+-------------+------------------+
+| PC4   | USART0_TX   | UART Console TX  |
++-------+-------------+------------------+
+| PC5   | USART0_RX   | UART Console RX  |
++-------+-------------+------------------+
+
+System Clock
+============
+
+The MGM240S SiP is configured to use the 39 MHz internal oscillator.
+
+Serial Port
+===========
+
+The MGM240S SiP has one USART and two EUSARTs.
+USART0 is connected to the board controller and is used for the console.
+
+Programming and Debugging
+*************************
+
+The Arduino Nano Matter contains an SAMD11 with CMSIS-DAP, allowing flashing, debugging, logging, etc. over
+the USB port. Doing so requires a version of OpenOCD that includes support for the flash on the MG24
+MCU. Until those changes are included in stock OpenOCD, the version bundled with Arduino can be
+used, or can be installed from the `OpenOCD Arduino Fork`_. When flashing, debugging, etc. you may
```

Comment:
Same comment than: https://github.com/zephyrproject-rtos/zephyr/pull/88321#discussion_r2037046606. BTW, maybe the already opened PR is sufficient: https://github.com/zephyrproject-rtos/openocd/pull/70. In this case, just add a comment explaining it is also useful for this board.